### PR TITLE
Arm64/SVE: JitStress/JitStressRegs fixes

### PR DESCRIPTION
**Pull request by kunalspathak**
- For FMA, handle case for FMA where `falseReg == embMaskOp1Reg`
- Workaround around `mov` when used as alias for `sel` because predicateRegister/vectorRegister are same
- When intrinsic is wrapped in `ConditionalSelect`, check the intrinsic flag if it needs low register mask
- Various APIs had missing flag for `HW_Flag_LowMaskedOperation`

There are still some functional failures with JitStress/JitStressRegs, but wanted to send this out.

### CODE REVIEW

**Comment by kunalspathak** [comment]
@dotnet/arm64-contrib

File: src/coreclr/jit/emitarm64sve.cpp [hunk 1]
```diff
@@ -3774,18 +3774,22 @@ void emitter::emitInsSve_R_R_R(instruction     ins,
                 // ORR is an alias for MOV, and is always the preferred disassembly.
                 ins = INS_sve_orr;
             }
-            else if (isPredicateRegister(reg3) &&
+            // TODO-SVE: Remove check for (opt == INS_OPTS_SCALABLE_B) when predicate registers
+            //  are present.
+            else if ((opt == INS_OPTS_SCALABLE_B) && isPredicateRegister(reg3) &&
```
Comment (a74nh) [inline src/coreclr/jit/emitarm64sve.cpp:3779]: You are using `INS_OPTS_SCALABLE_B` to check that you have a predicate? What about if moving a byte vector? Maybe add another enum value instead?

File: src/coreclr/jit/hwintrinsiclistarm64sve.h [hunk 2]
```diff
@@ -17,10 +17,10 @@
...
-HARDWARE_INTRINSIC(Sve,           AddAcross,                                                        -1,      1,      true,  {INS_sve_saddv,      INS_sve_uaddv,      INS_sve_saddv,      INS_sve_uaddv,      INS_sve_saddv,      INS_sve_uaddv,      INS_sve_uaddv,      INS_sve_uaddv,      INS_sve_faddv,      INS_sve_faddv},   HW_Category_SIMD,                  HW_Flag_Scalable|HW_Flag_BaseTypeFromFirstArg|HW_Flag_EmbeddedMaskedOperation)
+HARDWARE_INTRINSIC(Sve,           AddAcross,                                                        -1,      1,      true,  {INS_sve_saddv,      INS_sve_uaddv,      INS_sve_saddv,      INS_sve_uaddv,      INS_sve_saddv,      INS_sve_uaddv,      INS_sve_uaddv,      INS_sve_uaddv,      INS_sve_faddv,      INS_sve_faddv},   HW_Category_SIMD,                  HW_Flag_Scalable|HW_Flag_BaseTypeFromFirstArg|HW_Flag_EmbeddedMaskedOperation|HW_Flag_LowMaskedOperation)
```
Comment (a74nh) [inline src/coreclr/jit/hwintrinsiclistarm64sve.h:23]: These are going to be annoying to keep spotting. Will stress testing always fail when this is wrong?

Comment (kunalspathak) [reply src/coreclr/jit/hwintrinsiclistarm64sve.h:23]: yes, stress test should catch this.

File: src/coreclr/jit/emitarm64sve.cpp (hunk 1 above)
Comment (kunalspathak) [reply src/coreclr/jit/emitarm64sve.cpp]: yes, I don't need this check anymore after I added `INS_SCALABLE_OPTS_PREDICATE_MERGE_MOV`

**Review by a74nh — APPROVED** [review]
LGTM now

**Review by TIHan — APPROVED** [review]
LGTM. Glad we are catching these.

**Pull request merged**